src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs

```diff
@@ -505,13 +505,10 @@ public void Dispose()
                 byte[] buffer = new byte[expected.Length];
                 await Assert.ThrowsAsync<HttpListenerException>(() => context.Request.InputStream.ReadAsync(buffer, 0, buffer.Length));
                 await Assert.ThrowsAsync<HttpListenerException>(() => context.Request.InputStream.ReadAsync(buffer, 0, buffer.Length));
-
-                // Closing a response from a closed client if no writing has failed should fail.
-                Assert.Throws<HttpListenerException>(() => context.Response.Close());
```

Comment:
Yeah, good point, it's not going to change on Windows.

That said, I do think it's weird behavior, and it's not clear that matching the Windows behavior is worthwhile.